lib/ansible/plugins/action/__init__.py

```diff
@@ -763,7 +762,7 @@ def _execute_module(self, module_name=None, module_args=None, tmp=None, task_var
 
     def _remove_internal_keys(self, data):
         for key in list(data.keys()):
-            if key.startswith('_ansible_') or key in C.INTERNAL_RESULT_KEYS:
+            if key.startswith('_ansible_') and key != '_ansible_parsed' or key in C.INTERNAL_RESULT_KEYS:
```

Comment:
Seems reasonable for a 2.3 targeted fix.

Long term (after 2.3) may be useful to add a constant for the keys that are whitelisted ( ie something like `and key not in C.INTERNAL_RESULT_KEYS_WHITELIST` )